### ISSUE TITLE
ConditionalHeaders does not send 304 when If-Modified-Since equals Last-Modified unless server time zone is GMT

### ISSUE THREAD

**Issue by daniel-hagen** [ktorio/ktor#476]
When the server has determined a resource's LastModifiedVersion#lastModified as a local date/time in a timezone other than GMT (e.g. when the server's default timezone is GMT+X and the resource is a static file)
it responds to requests containing an If-Modified-Since equal to the Last-Modified from its previous response with "200 OK" instead of "304 Not Modified".

An example for a static resource (which is not modified between requests) served from local file by a server with default time zone GMT+1:
```
GET /foo HTTP/1.1
[...]

HTTP/1.1 200 OK
Last-Modified: Fri, 13 Jul 2018 11:39:26 GMT
[...]

GET /foo HTTP/1.1
If-Modified-Since: Fri, 13 Jul 2018 11:39:26 GMT
[...]

HTTP/1.1 200 OK
Last-Modified: Fri, 13 Jul 2018 11:39:26 GMT
[...]
```
If the local timezone is GMT, this works as expected:
```
GET /foo HTTP/1.1
[...]

HTTP/1.1 200 OK
Last-Modified: Fri, 13 Jul 2018 11:39:26 GMT
[...]

GET /foo HTTP/1.1
If-Modified-Since: Fri, 13 Jul 2018 11:39:26 GMT
[...]

HTTP/1.1 304 Not Modified
[...]
```
Looking at the code, the problem seems to be the date/time comparison in io.ktor.content.LastModifiedVersion#check at io/ktor/content/Versions.kt:87:
```
override fun check(call: ApplicationCall): VersionCheckResult {
    [...]

    if (ifModifiedSince != null) {
        if (normalized <= ifModifiedSince) {
    [...]
```
java.time.chrono.ChronoZonedDateTime#compareTo doesn't consider two instances representing the same point in time equal if their time zone differs.

The problem can be reproduced by setting io.ktor.tests.server.features.LastModifiedTest#date to ZonedDateTime.now(ZoneId.of("GMT+1")) (or any other non-GMT timezone)

**Comment by NorbertSandor** [comment]
I think this is a blocking problem for a production application...

A workaround if you can change the timezone: `TimeZone.setDefault(TimeZone.getTimeZone("GMT"))`

**Comment by cy6erGn0m** [comment]
Fixed in 1.1.0
